src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { HTMLAttributes, useRef, useEffect, useState } from "react";
+import styled from "@emotion/styled";
+import DropDownOption, { Props as OptionProps } from "components/dropDown/items/DropDownOption";
+import DropDownIcon from "components/dropDown/items/DropDownIcon";
+import DropUpIcon from "components/dropDown/items/DropUpIcon";
+import { TSelectedOption } from "components/dropDown/types/TSelectedOption";
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  name?: string;
+  label?: string;
+  children: React.ReactNode;
+  onSelectValueChange?: (value: string) => void;
+}
+
+const DropDown = ({ children, label, onSelectValueChange }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState<TSelectedOption | null>(null);
+  const dropDownRef = useRef<HTMLDivElement | null>(null);
+
+  const toggleDropdown = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const handleOptionClick = (value: string, text: React.ReactNode, index: number) => {
+    const newSelectedOption: TSelectedOption = {
+      index: index,
+      text: text,
+      value: value,
+    };
```

Comment:
사소한 것이지만, 객체에서 키값 이름과 키값의 값 (변수) 가 동일하면 한번만 써주어도 됩니다..!

```suggestion
    const newSelectedOption: TSelectedOption = { index, text, value };
```

Reply:
오.. 넵 반영하였습니다!